### DISCUSSION TITLE
Should Akkling libraries be rebuilt with Akka.NET 1.5?

### DISCUSSION THREAD

**Discussion by object** [akkadotnet/akka.net#6499] [General]
We are using Akka.NET in F# together with Akkling library. Until now Akkling never needed an updated for newer Akka.NET packages. It's just a thin wrapper offering F# API to access Akka.NET features. But after updating Akka.NET packages to the version 1.5, many of our tests failed with MissingMethodException.

There are two main types of failures: 
1. System.MissingMethodException: Method not found: 'Void Akka.Event.ILoggingAdapter.Log(Akka.Event.LogLevel, System.String, System.Object[])

2. System.MissingMethodException: Method not found: 'Void Akka.TestKit.TestKitBase.ExpectNoMsg(System.TimeSpan)'.

I know that both Logging and TestKit had some changes in the version 1.5. Does it mean that Akkling needs a rebuild?

**Comment by object** [comment]
Answering my own question: yes, Akkling requires rebuild with a reference to Akka.NET 1.5 libraries. I've opened PR at Akkling repo:
https://github.com/Horusiath/Akkling/pull/164

**Reply by Aaronontheweb** [reply]
As far as I know, these should be the only breaking changes Akka.NET v1.5 introduces - I explained some of the rationale for that here "[Akka NET v1.5 New Features and Upgrade Guide (7:27)](https://youtu.be/-UPestlIw4k?t=447)"

**Reply by object** [reply]
Just watched the video. It's clear that ILoggingAdapter.Log needs an update. Not clear why ExpectNoMsg was reported as missing method. But rebuilding Akkling with new Akka solved the problem.
